test/testactivitydata.cpp

```diff
@@ -87,6 +88,55 @@ class TestActivityData : public QObject
         QTest::addRow("data") << testData << fileFormat << mimeType << objectType << subject << path << fileName << activityType << activityId << message << objectName << link << datetime << icon << subjectRichString << subjectRichData << previewUrl;
     }
 
+    void createCalendarActivityJsonData(QString name, QString event, QString calendar)
+    {
+        const auto objectType = QStringLiteral("calendar");
+        const auto subject = QStringLiteral("%1 updated event %2 in calendar %3").arg(name, event, calendar);
+        const auto activityType = QStringLiteral("calendar_event");
+        const auto activityId = 10000;
+        const auto datetime = QDateTime::currentDateTime().toString(Qt::ISODate);
+        const auto icon = account->url().toString().append(QStringLiteral("/core/img/places/calendar.svg"));
+        const auto eventLink = QStringLiteral("/apps/calendar/dayGridMonth/now/edit/sidebar/A12bcD12AbcDEFgH456/next");
+
+        const auto subjectRichString = QStringLiteral("{actor} updated event {event} in calendar {calendar}");
+        const QJsonArray subjectRichData({
+            subjectRichString,
+            {{
+                {objectType, {{
+                    {QStringLiteral("type"), objectType},
+                    {QStringLiteral("id"), QStringLiteral()},
+                    {QStringLiteral("link"),  QStringLiteral()},
+                    {QStringLiteral("name"), calendar},
+                }}},
+                {QStringLiteral("event"), {{
```

Comment:
if I understand it correctly from @nickvergessen comment, this could also be event1
could you add tests that cover more cases to check the code is still adding the correct link ?